substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -2001,6 +2003,52 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Remove a nominator from a validator's nominators when the validator is blocked.
+		///
+		/// This function allows validators who have declared themselves as blocked (not accepting
+		/// nominations) to remove nominators who had previously nominated them.
+		///
+		/// The dispatch origin for this call must be _Signed_ by the validator controller.
+		///
+		/// - `who`: A list of nominator stash accounts that should be removed from nominating this
+		///   validator.
+		#[pallet::call_index(22)]
```

Comment:
please add this to the end, this new index will break all the rest and we should avoid that :)

Reply:
> please add this to the end, this new index will break all the rest and we should avoid that :)

just for my knowledge, whenever we update a pallet with a nex extrinsic it should be added to the end and not shift everything ? Like with smart contracts where shifting the storage order during an upgrade would break the storage state ?